mods/sib-plusplus-tweaker.wh.cpp

```diff
@@ -0,0 +1,453 @@
+}
+
+using SetWindowPos_t = decltype(&SetWindowPos);
+SetWindowPos_t SetWindowPos_orig = nullptr;
+BOOL WINAPI SetWindowPos_hook(
+    HWND hWnd,
+    HWND hWndInsertAfter,
+    int  X,
+    int  Y,
+    int  cx,
+    int  cy,
+    UINT uFlags
+)
+{
+    void *retaddr = __builtin_return_address(0);
+    if (((ULONGLONG)retaddr >= (ULONGLONG)g_hStartIsBackModule) && ((ULONGLONG)retaddr < ((ULONGLONG)g_hStartIsBackModule + g_StartIsBackSize)))
+    {
+        WCHAR className[MAX_PATH];
+        GetClassName(hWnd, className, MAX_PATH);
+        if(0 == wcscmp(className, L"NamespaceTreeControl"))
+        {
+            HDC dc = GetDC(hWnd);
+            return SetWindowPos_orig(
+                hWnd,
+                hWndInsertAfter,
+                X,
+                Y,
+                cx,
+                MulDiv(374, GetDeviceCaps(dc, LOGPIXELSY), 96),
+                uFlags
+            );
+        }
+    }
+    return SetWindowPos_orig(
+        hWnd,
+        hWndInsertAfter,
+        X,
+        Y,
+        cx,
+        cy,
+        uFlags
+    );
+}
+
+using InflateRect_t = decltype(&InflateRect);
+InflateRect_t InflateRect_orig = nullptr;
+BOOL WINAPI InflateRect_hook(
+    LPRECT lprc,
+    int    dx,
+    int    dy
+)
+{
+    void *retaddr = __builtin_return_address(0);
+    if (((ULONGLONG)retaddr >= (ULONGLONG)g_hStartIsBackModule) && ((ULONGLONG)retaddr < ((ULONGLONG)g_hStartIsBackModule + g_StartIsBackSize)))
+    {
+        //Wh_Log(L"retaddr = %p", retaddr);
+        uint8_t* ptr = (uint8_t*)(FindPattern("33 D2 48 8B CE 44 8D 42 FF FF 15 72 98 03 00",(uintptr_t)g_hStartIsBackModule));
+
+        if ((ULONGLONG)retaddr >= (ULONGLONG)(ptr) && (ULONGLONG)retaddr <= ((ULONGLONG)ptr + (ULONGLONG)16))
+        {
+            return TRUE;
+        }
+    }
+
+    return InflateRect_orig(
+        lprc,
+        dx,
+        dy
+    );
+}
+
+void LoadSettings(void)
+{
+    mod_settings.SIBPath = Wh_GetStringSetting(L"SIBPath");
+    mod_settings.DHTracking = Wh_GetIntSetting(L"DHTracking");
+    mod_settings.DisableCDSB = Wh_GetIntSetting(L"DisableCDSB");
+    mod_settings.DisableSMAnim = Wh_GetIntSetting(L"DisableSMAnim");
+    mod_settings.DisableCustomOrb = Wh_GetIntSetting(L"DisableCustomOrb");
+    mod_settings.ForceDWM = Wh_GetIntSetting(L"ForceDWM");
+    mod_settings.MatchSevenFolders = Wh_GetIntSetting(L"MatchSevenFolders");
+    mod_settings.RestoreAPPadding = Wh_GetIntSetting(L"RestoreAPPadding");
+}
+
+void CheckForStartIsBack()
+{
+    const wchar_t *SIB_PATH = mod_settings.SIBPath;
+    wchar_t workingSibPath[1024] = { 0 };
+    ExpandEnvironmentStringsW(SIB_PATH, workingSibPath, 1024);
+
+    g_hStartIsBackModule = LoadLibraryW(workingSibPath);
+    MODULEINFO miSib = { 0 };
+    if (!GetModuleInformation(
+        GetCurrentProcess(),
+        g_hStartIsBackModule,
+        &miSib,
+        sizeof(MODULEINFO)
+    ))
+    {
+        Wh_Log(L"Failed to get size of StartIsBack module! Exiting.");
+        return;
```

Comment:
I'd expect Wh_ModInit to return FALSE in this case.

Reply:
Converted void into a BOOL to handle it. Done.